FrontEnd/src/components/CompanyCard/CompanyCard.jsx

```diff
@@ -0,0 +1,209 @@
+
+  const { user } = useUser();
+  const ownProfile = user && user.id === profile.personId;
+
+  async function sendRequest(url, { arg: data }) {
+    const authToken = localStorage.getItem('Token');
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Token ${authToken}`,
+      },
+      body: JSON.stringify(data),
+    }).then();
+  }
+
+  const { trigger } = useSWRMutation(
+    `${process.env.REACT_APP_BASE_API_URL}/api/saved-list/`,
+    sendRequest
+  );
+
+  const handleClick = async () => {
+    try {
+      await trigger(
+        { company_pk: profile.id },
+        { optimisticData: () => setIsSaved(!isSaved) }
+      );
+      mutate(
+        (key) => typeof key === 'string' && key.startsWith('/api/profiles/'),
+        {
+          revalidate: true,
+        }
+      );
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const filledStar = (
+    <StarFilled
+      className={styles['star']}
+      onClick={handleClick}
+      data-testid="star"
+    />
+  );
+  const outlinedStar = (
+    <StarOutlined
+      className={styles['star']}
+      onClick={handleClick}
+      data-testid="emptystar"
+    />
+  );
+
+  const CategoryBadges = ({ categories }) => {
+    return (
+      <div>
+        {categories
+          ? categories.map((category) => (
+              <Badge
+                title=""
+                key={category.id}
+                size="medium"
+                count={category.name.toUpperCase()}
+                style={{
+                  backgroundColor: '#1F9A7C',
+                  fontWeight: 600,
+                  fontFamily: 'Inter',
+                  fontSize: 10,
+                  margin: 5,
+                }}
+              />
+            ))
+          : ''}
+      </div>
+    );
+  };
+
+  return (
+    <div className={styles['company-card']}>
+      <div className={styles['company-card__block']}>
+        <div className={styles['company-card__image-frame']}>
+          {profile.banner ? (
+            <img
+              src={profile.banner}
+              alt="Company Banner"
+              className={styles['company-card__image']}
+            />
+          ) : (
+            <img
+              className={styles['company-card__empty-image']}
+              src={`${process.env.REACT_APP_PUBLIC_URL}/svg/profile-view-image-empty.svg`}
+              alt={profile.name}
+            />
+          )}
+        </div>
+        <div className={styles['company-card__text-block']}>
+          <Tooltip
+            title={
+              profile.activities &&
+              profile.activities.map((element) => element.name).join(', ')
+            }
+            placement="bottom"
+            pointAtCenter={true}
+          >
+            <div className={styles['company-card__category-text']}>
+              {profile.activities &&
+                profile.activities
+                  .slice(0, 3)
```

Comment:
Why 3? Should we show the rest elements in the tooltip? Could we avoid using magic numbers and provide meaningful name for it

Reply:
Thank your for feedback. Done.